core/server/worker/src/main/java/alluxio/worker/block/CacheRequestManager.java

```diff
@@ -106,7 +111,43 @@ public void submitRequest(AsyncCacheRequest request) {
     } catch (Exception e) {
       LOG.warn("Failed to submit async cache request. request: {}", request, e);
       ASYNC_CACHE_FAILED_BLOCKS.inc();
-      mPendingRequests.remove(blockId);
+      mPendingAsyncRequests.remove(blockId);
+    }
+  }
+  // TODO(jianjian): return error message
+  /**
+   * Handles a request to cache a block synchronously.
+   *
+   * @param request the cache request fields will be available
+   * @return request succeed or not
+   */
+  public boolean submitRequest(CacheRequest request) {
```

Comment:
This is almost identical to `submitAsyncRequest()`.
By merging these two into a single method, we could avoid lots of duplication.
I recommend merging these to all the way up to the `BlockWorker` interface.